lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/TestFailUpdateEvaluator.cs

```diff
@@ -0,0 +1,238 @@
+                [Parameter(typeof(TaskConfigurationOptions.Identifier))] string taskId,
+                [Parameter(typeof(FailureType))] int failureType)
+            {
+                _maxIters = maxIters;
+                _iterations = 0;
+                _dim = dim;
+                _intArr = new int[_dim];
+                _workers = numWorkers;
+                _taskId = taskId;
+                _failureType = failureType;
+                Logger.Log(Level.Info, "TestUpdateFunction: TaskId: {0}", _taskId);
+                Logger.Log(Level.Info, "Failure type: {0} failure", _failureType == 0 ? "evaluator" : "task");
+            }
+
+            /// <summary>
+            /// Update function
+            /// </summary>
+            /// <param name="input">integer array</param>
+            /// <returns>The same integer array</returns>
+            UpdateResult<int[], int[]> IUpdateFunction<int[], int[], int[]>.Update(int[] input)
+            {
+                if (input[0] != (_iterations + 1) * _workers)
+                {
+                    Exceptions.Throw(new Exception("Expected input to update functon not same as actual input"), Logger);
+                }
+
+                _iterations++;
+                Logger.Log(Level.Info, "Received value {0} in iteration {1}", input[0], _iterations);
+                MakeException();
+
+                if (_iterations < _maxIters)
+                {
+                    for (int i = 0; i < _dim; i++)
+                    {
+                        _intArr[i] = _iterations + 1;
+                    }
+
+                    return UpdateResult<int[], int[]>.AnotherRound(_intArr);
+                }
+
+                return UpdateResult<int[], int[]>.Done(input);
+            }
+
+            /// <summary>
+            /// Initialize function. Sends integer array with value 1 to all mappers
+            /// </summary>
+            /// <returns>Map input</returns>
+            UpdateResult<int[], int[]> IUpdateFunction<int[], int[], int[]>.Initialize()
+            {
+                for (int i = 0; i < _dim; i++)
+                {
+                    _intArr[i] = _iterations + 1;
+                }
+
+                return UpdateResult<int[], int[]>.AnotherRound(_intArr);
+            }
+
+            private void MakeException()
+            {
+                if (_iterations == 10 && !_taskId.EndsWith("-" + NumberOfRetry))
+                { 
+                    Logger.Log(Level.Warning, "Simulating {0} failure for taskId {1}",
+                        _failureType == 0 ? "evaluator" : "task",
+                        _taskId);
+                    if (_failureType == 0)
+                    {
+                        // simulate evaluator failure
+                        Environment.Exit(1);
+                    }
+                    else
+                    {
+                        // simulate task failure
+                        throw new ArgumentNullException();
```

Comment:
Please add the exception message describing that this exception simulates failure.